.github/workflows/installer-win.yml

```diff
@@ -23,31 +28,35 @@ on:
     tags:
       - v*pre*
 
+  workflow_dispatch:
+
 concurrency:
   group: installer-win-${{ github.ref }}
   cancel-in-progress: true
 
 name: Create Windows Installer
 
 env:
-  IS_PRE: ${{ github.event_name == 'push' && github.ref_type == 'tag' }}
+  IS_PRE: ${{ contains(fromJSON('["workflow_dispatch", "push"]'), github.event_name) || (github.event.action == 'labeled' && github.event.label.name == 'build-installer-artifacts') }}
+  IS_STANDARD_PR: ${{ github.event_name == 'pull_request' && github.event.action != 'labeled' }}
 
 jobs:
   build:
     name: Windows installer
     runs-on: windows-latest
+    if: github.event.action != 'labeled' || github.event.label.name == 'build-installer-artifacts'
     strategy:
       matrix:
         build_type: ['Lite', 'Full']
     env:
-      UNWANTED_PACKAGES: ${{github.event_name == 'pull_request' && 'pip spyder-kernels python-slugify Rtree QDarkStyle PyNaCl psutil' || 'pip python-slugify Rtree PyNaCl psutil' }}
+      UNWANTED_PACKAGES: ${{github.event_name == 'pull_request' && github.event.action != 'labeled' && 'pip spyder-kernels python-slugify Rtree QDarkStyle PyNaCl psutil' || 'pip python-slugify Rtree PyNaCl psutil' }}
```

Comment:
PRs can have any label, but we want the workflow to behave differently for different pull request actions.

Previously, this workflow triggered on the default `pull_request` types: `opened`, `reopened`, or `sychronize`. This trigger caused the workflow to build developer versions (not "release" versions). In the case of this particular line, this meant that `spyder-kernels` was included in the `UNWANTED_PACKAGES` variable. For "release" versions (i.e. non-PR triggers) `spyder-kernels` would not be included in this variable.

With this PR, we want the behavior of the workflow to remain the same regarding release-like and non-release-like versions, but the trigger now includes the `labeled` `pull_request` type. This changes the logic from `github.event_name == 'pull_request'` to `github.event_name == 'pull_request' && github.event.action != 'labeled'`.

But what about PRs that get labeled an arbitrary label? Well, we don't have to worry about that logic on this line because the `build` job will not run in this case due to line 47. `github.event.action != 'labeled'` on line 47 covers all the previous trigger cases; `github.event.label.name == 'build-installer-artifacts'` on line 47 then permits only the additional `labeled` trigger when the label is correct, i.e. arbitrary labels given to a PR will still trigger this workflow but the jobs will be skipped.

Unfortunately, unlike the `push.tag` trigger, there is no way to filter the workflow trigger on a label pattern, so it must be done at the job level.

Reply:
@ccordoba12, did this sufficiently answer your question?

Reply:
Yeah, it does. Sorry for the delay